From e38617258b1aafd04bce2cc373ea0e551c9da066 Mon Sep 17 00:00:00 2001
From: Alexander Kartashov <alekskartashov@parallels.com>
Date: Tue, 2 Jul 2013 17:19:52 +0400
Subject: [PATCH] arm: added ARM versions of the POSIX timer syscalls

Signed-off-by: Pavel Emelyanov <xemul@parallels.com>
Signed-off-by: Andrey Vagin <avagin@openvz.org>
---
 arch/arm/syscall.def | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm/syscall.def b/arch/arm/syscall.def
index 39b578e..ec7301b 100644
--- a/arch/arm/syscall.def
+++ b/arch/arm/syscall.def
@@ -73,6 +73,11 @@ gettid				186	224	(void)
 futex				202	240	(u32 *uaddr, int op, u32 val, struct timespec *utime, u32 *uaddr2, u32 val3)
 set_tid_address			218	256	(int *tid_addr)
 restart_syscall			219	0	(void)
+timer_create			222	257	(clockid_t which_clock, struct sigevent *timer_event_spec, timer_t *created_timer_id)
+timer_settime			223	258	(timer_t timer_id, int flags, const struct itimerspec *new_setting, struct itimerspec *old_setting)
+timer_gettime			224	259	(int timer_id, const struct itimerspec *setting)
+timer_getoverrun		225	260	(int timer_id)
+timer_delete			226	261	(timer_t timer_id)
 exit_group			231	248	(int error_code)
 set_robust_list			273	338	(struct robust_list_head *head, size_t len)
 get_robust_list			274	339	(int pid, struct robust_list_head **head_ptr, size_t *len_ptr)
-- 
1.8.3.1

