From f9f591b867abc61b414b4bee84ac683860c89ae6 Mon Sep 17 00:00:00 2001
From: Radostin Stoyanov <radostin@redhat.com>
Date: Sun, 5 Sep 2021 21:38:06 +0100
Subject: [PATCH 011/249] coredump: drop unused variable

Signed-off-by: Radostin Stoyanov <radostin@redhat.com>
---
 coredump/criu_coredump/coredump.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/coredump/criu_coredump/coredump.py b/coredump/criu_coredump/coredump.py
index a9a8bb27c..42c2fed19 100644
--- a/coredump/criu_coredump/coredump.py
+++ b/coredump/criu_coredump/coredump.py
@@ -834,8 +834,6 @@ class coredump_generator:
 
         vmas = []
         for vma in mm["vmas"]:
-            size = self._get_vma_dump_size(vma)
-
             v = vma_class()
             v.filesz = self._get_vma_dump_size(vma)
             v.data = self._gen_mem_chunk(pid, vma, v.filesz)
-- 
2.35.1

