From 7ea1e500fb23ce96dc895c5d59aae5a3728040b0 Mon Sep 17 00:00:00 2001
From: Radostin Stoyanov <radostin@redhat.com>
Date: Sun, 5 Sep 2021 22:55:24 +0100
Subject: [PATCH 019/249] make: enable lint for coredump

Signed-off-by: Radostin Stoyanov <radostin@redhat.com>
---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 08761efed..1f0dec78b 100644
--- a/Makefile
+++ b/Makefile
@@ -409,11 +409,13 @@ lint:
 	flake8 --config=scripts/flake8.cfg test/others/rpc/config_file.py
 	flake8 --config=scripts/flake8.cfg lib/py/images/pb2dict.py
 	flake8 --config=scripts/flake8.cfg scripts/criu-ns
+	flake8 --config=scripts/flake8.cfg coredump/
 	shellcheck --version
 	shellcheck scripts/*.sh
 	shellcheck scripts/ci/*.sh scripts/ci/apt-install
 	shellcheck test/others/crit/*.sh
 	shellcheck test/others/libcriu/*.sh
+	shellcheck test/others/crit/*.sh test/others/criu-coredump/*.sh
 	shellcheck test/others/config-file/*.sh
 	# Do not append \n to pr_perror or fail
 	! git --no-pager grep -E '^\s*\<(pr_perror|fail)\>.*\\n"'
-- 
2.35.1

