From 0bef863f00df370b32c7c3288dbe1c805ad6905e Mon Sep 17 00:00:00 2001
From: Radostin Stoyanov <radostin@redhat.com>
Date: Tue, 7 Dec 2021 21:29:44 +0000
Subject: [PATCH 091/120] tls: fix typo

Signed-off-by: Radostin Stoyanov <radostin@redhat.com>
---
 criu/tls.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/criu/tls.c b/criu/tls.c
index 60bd105bc..6a7b523ac 100644
--- a/criu/tls.c
+++ b/criu/tls.c
@@ -8,7 +8,7 @@
 #include "cr_options.h"
 #include "xmalloc.h"
 
-/* Compatability with GnuTLS verson <3.5 */
+/* Compatability with GnuTLS version < 3.5 */
 #ifndef GNUTLS_E_CERTIFICATE_VERIFICATION_ERROR
 #define GNUTLS_E_CERTIFICATE_VERIFICATION_ERROR GNUTLS_E_CERTIFICATE_ERROR
 #endif
-- 
2.34.1

