From eb6062eef811fe062a7d2ce5745405b867559b05 Mon Sep 17 00:00:00 2001
From: Adrian Reber <areber@redhat.com>
Date: Fri, 3 Dec 2021 17:13:27 +0000
Subject: [PATCH 102/249] ci: update to latest Vagrant and Fedora images

Signed-off-by: Adrian Reber <areber@redhat.com>
---
 scripts/ci/vagrant.sh | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/scripts/ci/vagrant.sh b/scripts/ci/vagrant.sh
index 40c8416e1..4a4a16445 100755
--- a/scripts/ci/vagrant.sh
+++ b/scripts/ci/vagrant.sh
@@ -1,14 +1,14 @@
 #!/bin/bash
 
-# This script is used to run vagrant based tests on Travis.
-# This script is started via sudo from .travis.yml
+# This script is used to run vagrant based tests on Cirrus CI.
+# This script is started via .cirrus.yml
 
 set -e
 set -x
 
-VAGRANT_VERSION=2.2.16
-FEDORA_VERSION=34
-FEDORA_BOX_VERSION=34.20210423.0
+VAGRANT_VERSION=2.2.19
+FEDORA_VERSION=35
+FEDORA_BOX_VERSION=35.20211026.0
 
 setup() {
 	if [ -n "$TRAVIS" ]; then
-- 
2.35.1

