From 9a0c293c5ea8ca0430d1887b1f2647d0217e222f Mon Sep 17 00:00:00 2001
From: Radostin Stoyanov <rstoyanov@fedoraproject.org>
Date: Tue, 5 Apr 2022 10:48:40 +0100
Subject: [PATCH] fixup

Signed-off-by: Radostin Stoyanov <rstoyanov@fedoraproject.org>
---
 criu/cr-dump.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/criu/cr-dump.c b/criu/cr-dump.c
index 0984bd11c..4ca436d26 100644
--- a/criu/cr-dump.c
+++ b/criu/cr-dump.c
@@ -1057,7 +1057,7 @@ static int read_rseq_cs(pid_t tid, struct __ptrace_rseq_configuration *rseqc, st
 	 */
 	ret = ptrace_peek_area(tid, rseq, decode_pointer(rseqc->rseq_abi_pointer), sizeof(struct rseq));
 	if (ret) {
-		pr_err("ptrace_peek_area(%d, %lx, %lx, %lx): fail to read rseq struct\n", tid, (unsigned long)rseq,
+		pr_err("ptrace_peek_area(%d, %lx, %lx, %lu): fail to read rseq struct\n", tid, (unsigned long)rseq,
 		       (unsigned long)(rseqc->rseq_abi_pointer), sizeof(uint64_t));
 		return -1;
 	}
@@ -1067,7 +1067,7 @@ static int read_rseq_cs(pid_t tid, struct __ptrace_rseq_configuration *rseqc, st
 
 	ret = ptrace_peek_area(tid, rseq_cs, decode_pointer(rseq->rseq_cs.ptr64), sizeof(struct rseq_cs));
 	if (ret) {
-		pr_err("ptrace_peek_area(%d, %lx, %lx, %lx): fail to read rseq_cs struct\n", tid,
+		pr_err("ptrace_peek_area(%d, %lx, %lx, %lu): fail to read rseq_cs struct\n", tid,
 		       (unsigned long)rseq_cs, (unsigned long)rseq->rseq_cs.ptr64, sizeof(struct rseq_cs));
 		return -1;
 	}
-- 
2.35.1

