From 9086891f2e524f9630ee474b87e8b1530ff03c7a Mon Sep 17 00:00:00 2001
From: Radostin Stoyanov <rstoyanov@fedoraproject.org>
Date: Thu, 19 May 2022 08:45:46 +0100
Subject: [PATCH] config: Always use mntns-compat-mode

Signed-off-by: Radostin Stoyanov <rstoyanov@fedoraproject.org>
---
 criu/config.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/criu/config.c b/criu/config.c
index 14a11f9c3..f4a7f57c3 100644
--- a/criu/config.c
+++ b/criu/config.c
@@ -1105,10 +1105,9 @@ int check_options(void)
 	}
 #endif
 
-	if (opts.mntns_compat_mode && opts.mode != CR_RESTORE) {
-		pr_err("Option --mntns-compat-mode is only valid on restore\n");
-		return 1;
-	} else if (!opts.mntns_compat_mode && opts.mode == CR_RESTORE) {
+	// FIXME: Always use mntns-compat-mode until we have a proper fix for
+	// https://github.com/opencontainers/runc/pull/3442
+	if (opts.mode == CR_RESTORE) {
 		if (check_mount_v2()) {
 			pr_debug("Mount engine fallback to --mntns-compat-mode mode\n");
 			opts.mntns_compat_mode = true;
-- 
2.35.3

