diff --git a/include/ptrace.h b/include/ptrace.h
index 46b6def..9420acd 100644
--- a/include/ptrace.h
+++ b/include/ptrace.h
@@ -19,16 +19,17 @@
 
 #ifndef PTRACE_PEEKSIGINFO
 #define PTRACE_PEEKSIGINFO      0x4209
+
+/* Read signals from a shared (process wide) queue */
+#define PTRACE_PEEKSIGINFO_SHARED       (1 << 0)
+#endif
+
 struct ptrace_peeksiginfo_args {
         __u64 off;	/* from which siginfo to start */
         __u32 flags;
         __u32 nr;	/* how may siginfos to take */
 };
 
-/* Read signals from a shared (process wide) queue */
-#define PTRACE_PEEKSIGINFO_SHARED       (1 << 0)
-#endif
-
 #ifndef PTRACE_GETREGSET
 # define PTRACE_GETREGSET	0x4204
 # define PTRACE_SETREGSET	0x4205
